### PR TITLE
Run service samples on CI gate with ubuntu

### PR DESCRIPTION
**Pull request by vhvb1989**
Adding a new CI gate for the internal pipeline ( nightly ) to run samples.

During CMake config/generation, we are writing the samples binaries in a txt within the /build folder ( same strategy we have for code coverage).  The CI gate reads the content of the file, depending on the service name and runs all the samples.

We use a bash script that checks the return code of each sample and exits as soon as one of the samples fails.

Added:
 Key Vault ( All samples from all packages )
 Storage ( All samples from all packages )

### CODE REVIEW

**Comment by vhvb1989** [comment]
Running storage samples:

image

**Comment by vhvb1989** [comment]
Running KeyVault samples:

image

**Review by Jinming-Hu — DISMISSED** [review]
wait for storage team's review

**Comment by Jinming-Hu** [comment]
Please wait until https://github.com/Azure/azure-sdk-for-cpp/pull/3113 is merged

**Comment by vhvb1989** [comment]
> Please wait until #3113 is merged

I reverted storage changes as the current changes won't work after the changes you are doing to the samples. We can update your PR after merging this one for your samples to be run on CI or I can create a new PR later.

File: sdk/keyvault/azure-security-keyvault-keys/test/samples/sample1-hello-world/sample1_hello_world.cpp
Comment (antkmsft) [inline sdk/keyvault/azure-security-keyvault-keys/test/samples/sample1-hello-world/sample1_hello_world.cpp:33]: In samples at least, and in future in all code, let's use the `const X` instead of `X const`.
In this PR, I've seen instances of `auto const` and `uint8_t const` in samples, I recommend to update them.

File: sdk/keyvault/azure-security-keyvault-keys/test/samples/sample5-sign-verify/sample5_sign_verify.cpp
Comment (antkmsft) [inline sdk/keyvault/azure-security-keyvault-keys/test/samples/sample5-sign-verify/sample5_sign_verify.cpp:131]: `X <-> const`

**Review by antkmsft — APPROVED** [review]

File: sdk/keyvault/azure-security-keyvault-keys/test/samples/sample1-hello-world/sample1_hello_world.cpp
Comment (vhvb1989) [reply sdk/keyvault/azure-security-keyvault-keys/test/samples/sample1-hello-world/sample1_hello_world.cpp:33]: really?   why?   :(   

I would expect any user to know that it's the same.

**Pull request merged**
